x-pack/plugins/stack_connectors/public/connector_types/es_index/es_index_params.tsx

```diff
@@ -45,10 +46,17 @@ export const IndexParamsFields = ({
   );
   const [isActionConnectorChanged, setIsActionConnectorChanged] = useState<boolean>(false);
 
-  const getDocumentToIndex = (doc: Array<Record<string, any>> | undefined) =>
-    doc && doc.length > 0 ? (doc[0] as unknown as string) : undefined;
+  const getDocumentToIndex = (docs: Array<Record<string, any>> | undefined) => {
+    return docs && docs.length > 0
```

Comment:
How about we leave a comment explaining each operation here?

Reply:
Added comments!